drivers/adc/max9611pmb/max9611.c

```diff
@@ -0,0 +1,494 @@
+}
+
+/***************************************************************************//**
+ * @brief Get MAX9611 configured delay time
+ * @param dev 		  - The device structure.
+ * @param dtime	      - Pointer to store delay time read
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_get_delay(struct max9611_dev *dev, enum max9611_delay_time *dtime)
+{
+	int ret;
+	uint8_t temp;
+
+	ret = max9611_read(dev, MAX9611_CTR2, 1, &temp);
+	if (ret)
+		return ret;
+
+	*dtime = no_os_field_get(MAX9611_DTIME_MASK, temp);
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Configures MAX9611 retry time
+ * @param dev 		  - The device structure.
+ * @param rtime		  - Retry time option
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_set_retry(struct max9611_dev *dev, enum max9611_retry_time rtime)
+{
+	if (rtime < MAX9611_50MS || rtime > MAX9611_10MS)
+		return -EINVAL;
+
+	rtime = no_os_field_prep(MAX9611_RTIME_MASK, rtime);
+
+	return max9611_reg_update(dev, MAX9611_CTR2, rtime,
+				  MAX9611_RTIME_MASK);
+}
+
+/***************************************************************************//**
+ * @brief Get MAX9611 configured retry time
+ * @param dev 		  - The device structure.
+ * @param rtime	      - Pointer to store retry time read
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_get_retry(struct max9611_dev *dev, enum max9611_retry_time *rtime)
+{
+	int ret;
+	uint8_t temp;
+
+	ret = max9611_read(dev, MAX9611_CTR2, 1, &temp);
+	if (ret)
+		return ret;
+
+	*rtime = no_os_field_get(MAX9611_RTIME_MASK, temp);
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Receive data from ADC (single value only).
+ * @param dev 		  - The device structure.
+ * @param raw         - Pointer to store the raw 12-bit ADC reading
+ * @return            - 0 if successful, error otherwise.
+*******************************************************************************/
+int max9611_get_raw(struct max9611_dev *dev, uint16_t *raw)
+{
+	int ret;
+	uint8_t buf[2];
+
+	switch (dev->capture_type) {
+	case MAX9611_DATA_CSA:
+		ret = max9611_read(dev, MAX9611_CSA_MSB, 2, buf);
+		break;
+	case MAX9611_DATA_RSP:
+		ret = max9611_read(dev, MAX9611_RSP_MSB, 2, buf);
+		break;
+	case MAX9611_DATA_OUT:
+		ret = max9611_read(dev, MAX9611_OUT_MSB, 2, buf);
+		break;
+	case MAX9611_DATA_SET:
+		ret = max9611_read(dev, MAX9611_SET_MSB, 2, buf);
+		break;
+	case MAX9611_DATA_TMP:
+		ret = max9611_read(dev, MAX9611_TMP_MSB, 2, buf);
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	if (ret)
+		return ret;
+
+	*raw = no_os_get_unaligned_be16(buf);
+	*raw >>= 4;
```

Comment:
nit: we can define a mask here and use no_os_field_get

Reply:
defined macro MAX9611_RAW_DATA_MASK using NO_OS_GENMASK(15,4) and rewrote the raw value to "*raw = no_os_field_get(MAX9611_RAW_DATA_MASK, no_os_get_unaligned_be16(buf))"